app/controllers/signup.js

```diff
@@ -0,0 +1,194 @@
+import { action, set } from '@ember/object';
+
+export default class SignupController extends Controller {
+  @tracked isSubmitDisabled = true
+  @tracked title = 'Account Details'
+  @tracked formData = {
+    firstName: '',
+    lastName: '',
+    username: '',
+    email: '',
+    phoneNumber: '',
+    experience: '',
+    companyName: '',
+    designation: '',
+    github: '',
+    linkedIn: '',
+    twitter: '',
+  };
+
+  @tracked formErrors = {
+    firstName: false,
+    lastName: false,
+    username: false,
+    email: false,
+    phoneNumber: false,
+    experience: false,
+    companyName: false,
+    designation: false,
+    github: false,
+    linkedIn: false,
+    twitter: false,
+  };
+
+  @tracked fields = [
+    {
+      id: 'firstName',
+      label: 'First Name',
+      type: 'text',
+      errorMessage: 'First name is required',
+      required: true,
+      showError: false
+    },
+    {
+      id: 'lastName',
+      label: 'Last Name',
+      type: 'text',
+      errorMessage: 'Last name is required',
+      required: true,
+      showError: false
+    },
+    {
+      id: 'username',
+      label: 'Choose a username',
+      type: 'text',
+      errorMessage: 'Username is required',
+      required: true,
+      showError: false
+    },
+    {
+      id: 'email',
+      label: 'Email',
+      type: 'email',
+      errorMessage: 'Valid Email is required',
+      required: true,
+      showError: false,
+      validator: this.emailValidator
+    },
+    {
+      id: 'phoneNumber',
+      label: 'Phone Number',
+      type: 'string',
+      value: '+91-',
+      errorMessage: 'Enter a valid phone number',
+      required: false,
+      showError: false,
+      validator: this.phoneNumberValidator
+    },
+    {
+      id: 'experience',
+      label: 'Experience',
+      type: 'number',
+      errorMessage: 'Number of experience is required',
+      required: true,
+      showError: false
+    },
+    {
+      id: 'companyName',
+      label: 'Company Name ',
+      type: 'text',
+      errorMessage: '',
+      required: false,
+      showError: false
+    },
+    {
+      id: 'designation',
+      label: 'Designation ',
+      type: 'text',
+      errorMessage: '',
+      required: false,
+      showError: false
+    },
+    {
+      id: 'github',
+      label: 'Github ',
+      type: 'text',
+      errorMessage: 'GitHub username is required',
+      required: true,
+      showError: false
+    },
+    {
+      id: 'linkedIn',
+      label: 'LinkedIn ',
+      type: 'text',
+      errorMessage: 'LinkedIn username is required',
+      required: true,
+      showError: false
+    },
+    {
+      id: 'twitter',
+      label: 'Twitter ',
+      type: 'text',
+      errorMessage: 'Twitter handle is required',
+      required: true,
+      showError: false
+    },
+    {
+      id: 'website',
+      label: 'Website ',
+      type: 'text',
+      errorMessage: '',
+      required: false,
+      showError: false
+    },
+  ]
+
```

Comment:
Is there any way to refactor this?

Reply:
Sure, you can always create an issue or Raise a PR.
Can you please let us know what should be refactored?

Reply:
@takshch Can you clarify this?